From e573ad1526c14da82fea7e308cdf4761652877b2 Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Fri, 1 May 2020 11:48:16 +0800
Subject: [PATCH 61/81] HTVEC: Bit FIX2

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>
---
 drivers/irqchip/irq-loongson-htvec.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/irqchip/irq-loongson-htvec.c b/drivers/irqchip/irq-loongson-htvec.c
index 01bdc08f52..ae4398fb3a 100644
--- a/drivers/irqchip/irq-loongson-htvec.c
+++ b/drivers/irqchip/irq-loongson-htvec.c
@@ -45,7 +45,6 @@ static void htvec_irq_dispatch(struct irq_desc *desc)
 
 	for (i = 0; i < VEC_REG_COUNT; i++) {
 		pending = readl(priv->base + 4 * i);
-		writel(pending, priv->base + 4 * i);
 		while (pending) {
 			int bit = __ffs(pending);
 
@@ -68,7 +67,7 @@ static void htvec_ack_irq(struct irq_data *d)
 	void __iomem *addr = priv->base;
 
 	addr += VEC_REG_IDX(d->hwirq) * 4;
-	writel(VEC_REG_BIT(d->hwirq), addr);
+	writel(BIT(VEC_REG_BIT(d->hwirq)), addr);
 }
 
 static void htvec_mask_irq(struct irq_data *d)
-- 
2.31.1

