From ace22f0c96ff87926761d653db6594c372602fec Mon Sep 17 00:00:00 2001
From: handsomeyingyan <handsomeyingyan@github.com>
Date: Sat, 3 Oct 2020 18:00:34 +0800
Subject: [PATCH] fix

---
 arch/arm/boot/dts/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b7dd000e7..c24d57b50 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1132,8 +1132,8 @@ dtb-$(CONFIG_MACH_SUN9I) += \
 	sun9i-a80-optimus.dtb \
 	sun9i-a80-cubieboard4.dtb
 dtb-$(CONFIG_MACH_SUNIV) += \
-	suniv-f1c100s-licheepi-nano.dtb
-	suniv-widora-tiny200-v2.dtb
+	suniv-f1c100s-licheepi-nano.dtb \
+	suniv-widora-tiny200-v2.dtb \
 	suniv-widora-tiny200-v3.dtb
 dtb-$(CONFIG_ARCH_TANGO) += \
 	tango4-vantage-1172.dtb
-- 
2.28.0

