From ec3f13d1ec73800fd185cc701e7105467d0d5484 Mon Sep 17 00:00:00 2001
From: handsomeyingyan <handsomeyingyan@github.com>
Date: Sun, 28 Mar 2021 16:59:18 +0800
Subject: [PATCH] add csi camera support for v3s and s3

---
 arch/arm/boot/dts/sun8i-v3s.dtsi | 33 ++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-v3s.dtsi b/arch/arm/boot/dts/sun8i-v3s.dtsi
index f1568ed..a95dc1e 100644
--- a/arch/arm/boot/dts/sun8i-v3s.dtsi
+++ b/arch/arm/boot/dts/sun8i-v3s.dtsi
@@ -385,6 +385,26 @@
             interrupt-controller;
             #interrupt-cells = <3>;
 
+	   /omit-if-no-ref/
+	   csi0_mclk_pin: csi0-mclk-pin {
+		pins = "PE20";
+		function = "csi_mipi";
+		};
+
+	   /omit-if-no-ref/
+	   csi1_8bit_pins: csi1-8bit-pins {
+		pins = "PE0", "PE2", "PE3", "PE8", "PE9",
+			"PE10", "PE11", "PE12", "PE13", "PE14",
+		         "PE15";
+			function = "csi";
+		};
+
+	   /omit-if-no-ref/
+	   csi1_mclk_pin: csi1-mclk-pin {
+		pins = "PE1";
+		function = "csi";
+		};
+
             i2c0_pins: i2c0-pins {
                 pins = "PB6", "PB7";
                 function = "i2c0";
@@ -573,6 +593,19 @@
             #size-cells = <0>;
         };
 
+	csi1: camera@1cb4000 {
+		compatible = "allwinner,sun8i-v3s-csi";
+		reg = <0x01cb4000 0x3000>;
+		interrupts = <GIC_SPI 84 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&ccu CLK_BUS_CSI>,
+			 <&ccu CLK_CSI1_SCLK>,
+			 <&ccu CLK_DRAM_CSI>;
+		clock-names = "bus", "mod", "ram";
+		resets = <&ccu RST_BUS_CSI>;
+		status = "disabled";
+	};
+
+
         gic: interrupt-controller@1c81000 {
             compatible = "arm,gic-400";
             reg = <0x01c81000 0x1000>,
-- 
2.25.1

