From f6b9810d398b008d3347ba03b5551f64af3d9929 Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 042/131] Fix touchpad status error after STR/STD

Signed-off-by: Rui Wang <wangr@lemote.com>
Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/input/mouse/sentelic.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/mouse/sentelic.c b/drivers/input/mouse/sentelic.c
index e78c4c7eda34..e9eddf4e9854 100644
--- a/drivers/input/mouse/sentelic.c
+++ b/drivers/input/mouse/sentelic.c
@@ -899,8 +899,8 @@ static int fsp_activate_protocol(struct psmouse *psmouse)
 		pad->flags |= FSPDRV_FLAG_EN_OPC;
 
 		/* Enable on-pad vertical and horizontal scrolling */
-		fsp_onpad_vscr(psmouse, true);
-		fsp_onpad_hscr(psmouse, true);
+		fsp_onpad_vscr(psmouse, pad->vscroll);
+		fsp_onpad_hscr(psmouse, pad->hscroll);
 	} else {
 		/* Enable absolute coordinates output for Cx/Dx hardware */
 		if (fsp_reg_write(psmouse, FSP_REG_SWC1,
-- 
2.31.1

