From b73b360d4d8f3b31e6b15f6109bceb0a426cd452 Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 053/131] R8168: Enable WoL (Wake on LAN) by default

Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/net/ethernet/realtek/r8168/r8168.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/realtek/r8168/r8168.h b/drivers/net/ethernet/realtek/r8168/r8168.h
index b90f19fde296..2fc2b93f47f4 100644
--- a/drivers/net/ethernet/realtek/r8168/r8168.h
+++ b/drivers/net/ethernet/realtek/r8168/r8168.h
@@ -302,6 +302,9 @@ do { \
 #define true  1
 #endif
 
+#define ENABLE_S5WOL	1
+#define CONFIG_SOC_LAN	1
+
 //Hardware will continue interrupt 10 times after interrupt finished.
 #define RTK_KEEP_INTERRUPT_COUNT (10)
 
-- 
2.31.1

