From 4e640c43b4627e47688b07fc9975bb22b416f2a3 Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Sun, 28 May 2017 09:29:33 +0800
Subject: [PATCH 094/131] 8250_pci: Add a new PLX9050 serial port card support

Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/tty/serial/8250/8250_pci.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/drivers/tty/serial/8250/8250_pci.c b/drivers/tty/serial/8250/8250_pci.c
index 553bd9ec37c1..eb3b8b785803 100644
--- a/drivers/tty/serial/8250/8250_pci.c
+++ b/drivers/tty/serial/8250/8250_pci.c
@@ -92,6 +92,12 @@ setup_port(struct serial_private *priv, struct uart_8250_port *port,
 	if (bar >= PCI_NUM_BAR_RESOURCES)
 		return -EINVAL;
 
+	if ((dev->vendor == PCI_VENDOR_ID_PLX) &&
+	    (dev->device == PCI_DEVICE_ID_PLX_9050) &&
+	    (dev->subsystem_vendor == PCI_VENDOR_ID_PLX) &&
+	    (dev->subsystem_device == PCI_DEVICE_ID_PLX_9050))
+		offset += 0x80;
+
 	if (pci_resource_flags(dev, bar) & IORESOURCE_MEM) {
 		if (!pcim_iomap(dev, bar, 0) && !pcim_iomap_table(dev))
 			return -ENOMEM;
@@ -274,6 +280,12 @@ static int pci_plx9050_init(struct pci_dev *dev)
 		 * deep FIFOs
 		 */
 		irq_config = 0x5b;
+
+	if ((dev->vendor == PCI_VENDOR_ID_PLX) &&
+	    (dev->device == PCI_DEVICE_ID_PLX_9050) &&
+	    (dev->subsystem_vendor == PCI_VENDOR_ID_PLX) &&
+	    (dev->subsystem_device == PCI_DEVICE_ID_PLX_9050))
+		irq_config = 0x53;
 	/*
 	 * enable/disable interrupts
 	 */
@@ -2198,6 +2210,15 @@ static struct pci_serial_quirk pci_serial_quirks[] __refdata = {
 		.setup		= pci_default_setup,
 		.exit		= pci_plx9050_exit,
 	},
+	{
+		.vendor		= PCI_VENDOR_ID_PLX,
+		.device		= PCI_DEVICE_ID_PLX_9050,
+		.subvendor	= PCI_VENDOR_ID_PLX,
+		.subdevice	= PCI_DEVICE_ID_PLX_9050,
+		.init		= pci_plx9050_init,
+		.setup		= pci_default_setup,
+		.exit		= pci_plx9050_exit,
+	},
 	{
 		.vendor		= PCI_VENDOR_ID_PLX,
 		.device		= PCI_DEVICE_ID_PLX_9050,
@@ -4276,6 +4297,10 @@ static const struct pci_device_id serial_pci_tbl[] = {
 		PCI_SUBDEVICE_ID_UNKNOWN_0x1584, 0, 0,
 		pbn_b2_4_115200 },
 	/* Unknown card - subdevice 0x1588 */
+	{	PCI_VENDOR_ID_PLX, PCI_DEVICE_ID_PLX_9050,
+		PCI_VENDOR_ID_PLX,
+		PCI_DEVICE_ID_PLX_9050, 0, 0,
+		pbn_b2_8_921600 },
 	{	PCI_VENDOR_ID_PLX, PCI_DEVICE_ID_PLX_9050,
 		PCI_VENDOR_ID_PLX,
 		PCI_SUBDEVICE_ID_UNKNOWN_0x1588, 0, 0,
-- 
2.31.1

